mozjs/src/gc/root.rs

```diff
@@ -43,7 +43,17 @@ impl<'a, T: 'a + RootKind> RootedGuard<'a, T> {
     }
 
     pub fn handle_mut(&mut self) -> MutableHandle<T> {
-        unsafe { MutableHandle::from_marked_location(self.deref_mut()) }
+        unsafe { MutableHandle::from_marked_location(self.as_ptr()) }
+    }
+
+    pub fn as_ptr(&self) -> *mut T {
+        // SAFETY: self.root points to an inbounds allocation
+        unsafe { (&raw mut (*self.root).ptr).cast() }
+    }
```

Comment:
This ptr can become invalid if GC runs, right?

Reply:
No, I don't think so, happy to be proven wrong if I am though. This pointer should be equivalent [^1] to the `self.root` ptr, which should be either equivalent to or a parent of any pointer the GC uses, so the GC shouldn't invalidate it.

Miri seems to agree it isn't invalidated in [this toy example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=4222e47257fdcb7e6d7c8dbeb126f85d).

[^1]: Equivalent in the tree borrows model. In the same contiguous block of `SharedRW` tags in the stacked borrows model.

Reply:
Indeed, I got confuse with all indirection.